### ISSUE TITLE
train_sop_actor_critic.py and train_sop_mappo.py not converge

### ISSUE THREAD

**Issue by Yuxin916** [LijunSun90/pursuitFSC2#3]
Hi! Sorry for bothering again. 

I have run through few times of experiments when 
- world_size=6, 
- fov=11, 
- n_pursuers=4 
- n_evaders=1
- The evader is "DoNothingPrey". 
- n_epochs=1000, 
- steps_per_epoch=8000
- max_episode_length = 100
- other rl hyperparameter remain unchanged 

Under this most simplest case, the converge performance of actor_critic and mappo does not look good and the capture i could not tell is intentional or just randomly catch successful. The statistical result not good.

Could you provide the hyperparameter you used as your actor_critic and mappo demo? To provide more information, i would be appreciate if we can change contact and discuss further. Here is my email address: caiy0039@e.ntu.edu.sg. 

Again, thank you and best regards.

**Comment by LijunSun90** [comment]
Hi, 

1. The hyper parameters in the codes, such as train_sop_actor_critic.py and train_sop_mappo.py,  are exactly the ones that we used for the results in the paper.
2. As for the hyperparameters you provided, steps_per_epoch=8000 may be a little large because it means 8000 * 4 = 32000 training samples will be generated per epoch for the case of 4 pursuer.
4. n_epochs = 1000 is not enough for the surrounding based capture. Sorry, the last time I said 1000 epochs is sufficient is for the occupying-based capture, such as the pettingzoo pursuit environment, which is the self-orgainizing search task (multi_target_self_organizing_search) in our work.
5. In the multi_target_self_organizing_pursuit/data/log_actor_critic/pursuit_epoch2000x14.gif, and multi_target_self_organizing_pursuit/data/log_mappo/sop_mappo/test_sop_mappo.gif, we render our training results for A2C and MAPPO, for your reference. 
6. For your "not good result", we can discuss more. According to our experience, it is not easy to achieve perfect surrounding based capture strictly under the setup of POSG (partially observable stochastic game) where simultaneous decisions are made after the simultaneous observations of agents, especially in an implicit coordination way (no communication, no full observation). The perfect I mean here is in terms of the efficiency, obstacle avoidance, and multi-agent collision avoidance. Our current result is for your reference. And our current reward function may not be the best, although workable. 

We are pleased to be responsible for the reproductivity of our work and discuss related research problems.
You are welcome to contact us by: Lijun.Sun@student.uts.edu.au, 11860004@mail.sustech.edu.cn, or sunlijun522@126.com.

Best regards.

**Comment by Yuxin916** [comment]
Hi, 
Again thank you for your comprehensive answer. I really really appreciate that.
